### PR TITLE
@eessex => Mobile Login + Reset Password

### PR DESCRIPTION
**Pull request by kanaabe**
Flows for Login + Reset Password

- [x] Still needs a few style updates and tests

mobile-login

image

### CODE REVIEW

File: src/Components/Authentication/Mobile/LoginForm.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    }
+    actions.setFieldError("email", "Email does not exist.")
+    actions.setSubmitting(false)
+    return false
+  })
+}
+
+export const MobileLoginForm: FormComponentType = props => {
+  const steps = [
+    <Step validationSchema={LoginValidator.email} onSubmit={checkEmail}>
```
Comment (damassi) [inline src/Components/Authentication/Mobile/LoginForm.tsx]: I just love this wizard / step component; very nicely designed.

**Comment by kanaabe** [comment]
@eessex I made you the reviewer so you can get more acquainted with this but adding @xtina-starr here for more eyes too!

File: src/Components/Authentication/helpers.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import * as sharify from "sharify"
```
Comment (eessex) [inline src/Components/Authentication/helpers.ts]: Interesting, didn't realize we had sharify in reaction too -- does this pick up data from force's config?

File: src/Components/Authentication/helpers.ts [hunk 3]
```diff
@@ -34,3 +37,37 @@ export async function sendAuthData(
...
+    {
+      appToken: sharify.data.XAPP_TOKEN,
+    }
+  ).then(({ data }: any) => {
+    if (data.user.userAlreadyExists) {
+      if (shouldExist) {
+        return true
+      } else {
+        actions.setFieldError("email", "Email already exists.")
+        actions.setSubmitting(false)
```
Comment (eessex) [inline src/Components/Authentication/helpers.ts]: Curious to know where these actions come from -- are they stored in context?

File: src/Components/Authentication/helpers.ts (hunk 2 above)
Comment (kanaabe) [reply src/Components/Authentication/helpers.ts]: I assumed so but now that I look for it, I'm not sure if it'd actually work once in Force. @xtina-starr is working on integrating `MobileSignUp` into Force so we'll see if it's working as expected.

File: src/Components/Authentication/helpers.ts (hunk 3 above)
Comment (kanaabe) [reply src/Components/Authentication/helpers.ts]: These actions are passed down as args from Formik's `onSubmit` here: https://github.com/artsy/reaction/pull/750/files#diff-789fe1e619c39fe8bf9ce02131a25dddR26

**Pull request merged**